Story/quad 90 sprint2

rajataggarwal91 opened the pull request:
Created an about page in public website. About page reads a dummy string with bold and italics tags from controller and  prints it on the about page with formatted text.

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:
```diff
@@ -60,114 +58,132 @@ public IRetrieveProjectManager getProjectManager() {
\ No newline at end of file
+
+	@RequestMapping(value = "sites/{ProjectUnixName}/about", method = RequestMethod.GET)
+	public String showAbout(@PathVariable("ProjectUnixName") String unixName, Model model, Principal principal) throws QuadrigaStorageException {
```
this methdo should be in its own controller

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:
```diff
@@ -60,114 +58,132 @@ public IRetrieveProjectManager getProjectManager() {
+	@RequestMapping(value = "sites/{ProjectUnixName}/about", method = RequestMethod.GET)
+	public String showAbout(@PathVariable("ProjectUnixName") String unixName, Model model, Principal principal) throws QuadrigaStorageException {
+		IProject project = getProjectDetails(unixName);
+		String projectid = project.getProjectId();
```
if there is no project with that unix id, this will throw a null pointer exception

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteProjectController.java:
```diff
@@ -60,114 +58,132 @@ public IRetrieveProjectManager getProjectManager() {
+		model.addAttribute("project", project);
+		model.addAttribute("aboutProject", aboutProject);
+		return "sites/about";
+	}
```
use a different return string. There might be an about page for the whole public project part of Quadriga, which will be at "sites/about". Use maybe sites/project/about.

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/about.jsp:
```diff
@@ -1,12 +1,48 @@
+
+
+*/
+</script>
```
ARe you using the about page of Quadriga here? this is not rhe right page. You need to create your own one.

jdamerow replied:
I'm confused, isn't that still the main Quadriga about page? why do you change that one?

jdamerow commented:
I'll close this pull request for now. Just reopen it when you made the changes (rather than creating a new PR).

diging-jenkins commented:
Can one of the admins verify this patch?

diging-jenkins commented:
Can one of the admins verify this patch?

rajataggarwal91 commented:
Made changes according to comments. New controller and new jsp page for about page. Removed unnecessary error prone statements from controller. Changed the return string to return to /project/about

jdamerow commented:
@rajataggarwal91, why are you changing so many files?

rajataggarwal91 commented:
Now it shows 6 file changes instead of 60. Please review again. Apologies for inconvenience.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutController.java:
```diff
@@ -0,0 +1,43 @@
+	@Autowired
+	private IRetrieveProjectManager projectManager;
+
+	private IProject getProjectDetails(String name) throws QuadrigaStorageException {
```
I don't think you need this method, since it's just one line of code, just call the projectManager directly in your method (line 36).

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutController.java:
```diff
@@ -0,0 +1,43 @@
+		return projectManager.getProjectDetailsByUnixName(name);
+	}
+
+	public IRetrieveProjectManager getProjectManager() {
```
do you need those methods anywhere? If not, delete the getter and setter.

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/about.jsp:
```diff
@@ -1,12 +1,48 @@
+<html>
+<head>
+  	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+  	<title>About Project ${project} }</title>
```
Is there one '}' too many?

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/about.jsp:
```diff
@@ -1,12 +1,48 @@
+</head>
 
-<!-- Content -->
+<body>
```
this page should be part of a skeleton, so no html header or body tag required.

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/public/navbar.jsp:
```diff
@@ -1,4 +1,5 @@
-<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="${pageContext.servletContext.contextPath}/sites/${project.unixName}/networks">Explore</a></li>
-<li ${currentPage == "networksearch" ? "class=\"active\"" : ""}><a href="">Search</a></li>
\ No newline at end of file
+<li ${currentPage == "exploregraph" ? "class=\"active\"" : ""}><a href="">Explore</a></li>
```
Did you delete the link to the explore page accidentally or on purpose?

jdamerow commented on Quadriga/src/main/webapp/WEB-INF/views/public/publicwebsiteabout.jsp:
```diff
@@ -0,0 +1,16 @@
+ ${aboutProject }
+  </article> 
+</body>
+</html>
```
this page should be embedded in a skeleton, so no header/body tags necessary.

rajataggarwal91 commented:
Hi Julia.
I have made another commit with the changes you described above.
1. Removed getter, setter and extra function and called it directly
2. Not using about.jsp, reverted it back to "Under contruction page". Now using publicwebsiteabout.jsp.
3. The link was accidently deleted while merging. Now it should work.
4. All HTML header and body tags were removed from .jsp created.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutController.java:
```diff
@@ -0,0 +1,31 @@
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+
+@Controller
+public class WebsiteAboutController {
```
Needs javadoc

jdamerow replied:
Also, please adjust your eclipse settings to use 4 spaces:
https://diging.atlassian.net/wiki/pages/viewpage.action?pageId=8781898
and then reformat this class.

jdamerow commented:
ok to test

jdamerow commented:
cool, once the last two things are done this is ready to be merged.

rajataggarwal91 commented:
Both changes "javadocs" and "indentation formatting" were made. Please review.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/WebsiteAboutController.java:
```diff
@@ -0,0 +1,39 @@
+ * 
+ * @author Rajat Aggarwal
+ *
+ */
```
javadoc should go right before the class starts not before the imports

rajataggarwal91 commented:
Javadocs now after import statements.

jdamerow commented:
retest this please

jdamerow commented:
retest this please
